### ISSUE TITLE
Ensure that each test file passes separately

### ISSUE THREAD

**Issue by zenspider** [inspec/inspec#3996]
I want to ensure that all test files are individually runnable. I've seen some things that suggest that they aren't. I need to get green first, then do this, then start to attack the intermittent failures we've seen here and there.

**Comment by clintoncwolfe** [comment]
I believe the rake task `rake test:isolated` is intended to do exactly this - see https://github.com/inspec/inspec/blob/b4ae6cb342f6ed422ad2b67204c1775c0f24c1be/Rakefile#L74

**Comment by zenspider** [comment]
It only picks up a very small subset of the tests... 

```
1866 % ls test/unit/*_test.rb | wc -l
       8
1867 % find test/unit -name \*_test.rb | wc -l
     212
1868 % find test -name \*_test.rb | wc -l
     238
```

I'll patch that up (it also needs some basic simplification). Also that `-w` is gonna be a bad time (~2k lines of output per file).

I was also adding some automation during my process to keep track of which ones were bad so it was quicker to whittle down. Not sure if that fits in this task.

**Comment by zenspider** [comment]
Fixed in #4031
